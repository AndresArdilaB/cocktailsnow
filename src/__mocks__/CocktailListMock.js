const CocktailList = [{
  idDrink: '14029',
  strDrink: "'57 Chevy with a White License Plate",
  strDrinkThumb: 'https://www.thecocktaildb.com/images/media/drink/qyyvtu1468878544.jpg',
}, {
  idDrink: '178318',
  strDrink: '747 Drink',
  strDrinkThumb: 'https://www.thecocktaildb.com/images/media/drink/i9suxb1582474926.jpg',
}];

export default CocktailList;
