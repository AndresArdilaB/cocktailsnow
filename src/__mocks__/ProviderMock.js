import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import { testState } from '../frontend/initialState';
import reducer from '../frontend/reducers';

const store = createStore(reducer, testState);
const history = createBrowserHistory();

const ProviderMock = ({ children }) => (
  <Provider store={store}>
    <Router history={history}>
      {children}
    </Router>
  </Provider>
);

export default ProviderMock;
