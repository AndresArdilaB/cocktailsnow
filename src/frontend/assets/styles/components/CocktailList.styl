.container_list 
  width: 1065px
  margin: auto
  @media screen and (max-width: 1100px)
    width: 545px;
  @media screen and (max-width: 600px)
    width: 275px

.cocktails_list 
  display: grid
  grid-gap: 0.5rem
  grid-template-columns: 254px 254px 254px 254px
  @media screen and (max-width: 1100px)
    grid-template-columns: 254px 254px
  @media screen and (max-width: 600px)
    grid-template-columns: 254px