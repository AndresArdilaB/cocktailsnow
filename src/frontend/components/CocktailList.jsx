/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import CocktailListItem from './CocktailListItem';
import '../assets/styles/components/CocktailList.styl';

const CocktailList = ({ items }) => {

  return (
    <section className='container_list'>
      <div className='cocktails_list'>
        {
          items.map((d, i) => (
            <div id={d.idDrink} key={d.idDrink}>
              <CocktailListItem {...d} />
            </div>
          ))
        }
      </div>
    </section>
  );
};

export default CocktailList;
