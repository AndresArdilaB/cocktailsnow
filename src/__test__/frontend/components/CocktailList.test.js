/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { mount, shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import ProviderMock from '../../../__mocks__/ProviderMock';
import CocktailListMock from '../../../__mocks__/CocktailListMock';
import CocktailList from '../../../frontend/components/CocktailList';

describe('<CocktailList />', () => {

  test('Component render', () => {
    const cocktailList = shallow(
      <ProviderMock>
        <CocktailList />
      </ProviderMock>,
    );

    expect(cocktailList.length).toEqual(1);
  });

  test('Test cocktail list', () => {
    const wrapper = mount(
      <ProviderMock>
        <CocktailList items={CocktailListMock} />
      </ProviderMock>,
    );

    expect(wrapper.find('[id="14029"]')).toHaveLength(1);
  });

});

describe('CockailList Snapshot', () => {
  test('Testiong render component UI', () => {
    const cocktailList = create(
      <ProviderMock>
        <CocktailList items={CocktailListMock} />
      </ProviderMock>,
    );

    expect(cocktailList.toJSON()).toMatchSnapshot();
  });
});
