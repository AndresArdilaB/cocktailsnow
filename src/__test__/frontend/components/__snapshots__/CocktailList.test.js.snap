// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CockailList Snapshot Testiong render component UI 1`] = `
<section
  className="container_list"
>
  <div
    className="cocktails_list"
  >
    <div
      id="14029"
    >
      <div
        className="container_list_item"
      >
        <a
          className="link"
          href="/cocktail/14029"
          onClick={[Function]}
        >
          <div
            className="media_list_item"
          >
            <img
              alt="The Cocktel"
              src="https://www.thecocktaildb.com/images/media/drink/qyyvtu1468878544.jpg"
            />
          </div>
          <div>
            '57 Chevy with a White License Plate
          </div>
        </a>
      </div>
    </div>
    <div
      id="178318"
    >
      <div
        className="container_list_item"
      >
        <a
          className="link"
          href="/cocktail/178318"
          onClick={[Function]}
        >
          <div
            className="media_list_item"
          >
            <img
              alt="The Cocktel"
              src="https://www.thecocktaildb.com/images/media/drink/i9suxb1582474926.jpg"
            />
          </div>
          <div>
            747 Drink
          </div>
        </a>
      </div>
    </div>
  </div>
</section>
`;
